### ISSUE TITLE
Reload project does not seem to take any effect

### ISSUE THREAD

**Issue by rongcuid** [AdaCore/ada_language_server#1182]
When project file is modified (such as changing the list of main files), running the `Ada: Reload project` does not have any effect, such as updating the list of main files. Vscode must be reloaded for project file updates to take effect.

Environment: Mac OS 14.2.1; using alire to install Gnat native.

Reproduction:

1. Use `alr` to create a new binary project
2. Open the project with vscode
3. Build and run main
4. Rename the main file and update in the `.gpr` project definition
5. `Ada: Reload project`
6. Try build and run again

Expected:

* Build and run menu should update

Actual:

* Menu is still the main specified in stale definition

**Comment by sanyaade-teachings** [comment]
> When project file is modified (such as changing the list of main files), running the `Ada: Reload project` does not have any effect, such as updating the list of main files. Vscode must be reloaded for project file updates to take effect.
> 
> Environment: Mac OS 14.2.1; using alire to install Gnat native.
> 
> Reproduction:
> 
> 1. Use `alr` to create a new binary project
> 2. Open the project with vscode
> 3. Build and run main
> 4. Rename the main file and update in the `.gpr` project definition
> 5. `Ada: Reload project`
> 6. Try build and run again
> 
> Expected:
> 
> * Build and run menu should update
> 
> Actual:
> 
> * Menu is still the main specified in stale definition

Totally agree!: I have been wrangling with the same issues for almost 3 days now
I think I need another direction for the teachings of Ada

**Comment by eliericha** [comment]
Hello folks,

Thank you for reporting this issue. I have good news and bad news :)

The good news is that the issue has already been fixed last week and no longer occurs in the development version.

The bad news is that the fix went in right after we did the last release, v24.0.4. So currently I am able to reproduce the bug with the last release.

I will discuss with my colleagues about the possibility of performing another release with the fix soon.
